### PR TITLE
Save tailored setup steps on store

### LINKED ISSUE #66457: Stepper: Store setup step data in store
The processing step for the tailored flows is now at the end of the flow, not after the setup step (like the link in bio setup).

What we can do is store the values from the setup step in the store.
We'll then process it in another step at the end of the flow

Link to the figma: BcUdhLlhAp7UHtQHNTI4zA-fi-3147%3A78984

### PR DESCRIPTION
**Pull request by renancarvalho**
#### Proposed Changes

Store on redux store `siteTitle, siteDescription, siteAccentColor( newsletters only ) and siteImageLogo` for link in bio and newsletters flow

This is moved there to be used later on further step to persist data on the server.

- Pull this branch and run yarn start from the calypso root folder
- Navigate to http://calypso.localhost:3000/setup/linkInBioSetup?flow=link-in-bio&siteSlug=YOUR-SITE.wordpress.com
- Open or install [redux dev tools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en)
- After filling the fields, you should be able to see the siteTittle, siteDescription and SiteLogo saved there.
Similar to this:

image

Note, while implementing it I noticed that Redux does not like the idea of having Blob or Files in the Store, I implemented a workaround that basically transform the image into a base64string and then save it into the Store.
Reference -->  https://redux.js.org/style-guide/#do-not-put-non-serializable-values-in-state-or-actions

Related to #66457
Fixes #66457

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx [hunk 1]
```diff
@@ -138,7 +135,7 @@ const NewsletterSetup: Step = ( { navigation } ) => {
 						onChangeComplete={ ( value ) => setAccentColor( value.hex ) }
 					/>
 				</Popover>
-				<FormFieldset disabled={ isLoading }>
+				<FormFieldset disabled={ false }>
```
Comment (escapemanuele) [inline client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx]: We could have a boolean `isSubmitting` that becomes true and disables the input `onSubmit`!

**Review by escapemanuele — COMMENTED** [review]
Great job! Left a comment for you if you want to implement that 👍

**Review by escapemanuele — APPROVED** [review]
Looks good to me 🥇

File: client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx (hunk 1 above)
Comment (renancarvalho) [reply client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx]: Hi Emanuele =), I think we can just remove it completely once we are moving the submit part to another place.

**Pull request merged**